### ISSUE TITLE
Linebreaks not appropriately saved in logs

### ISSUE THREAD

**Issue by discardedaccount4299** [fcwill/fchat-pidgin#162]
When linebreaks are used in Pidgin's fchat, the pidgin portable plugin saves both a carriage return and linefeed. However, Pidgin's logs are saved in HTML format thus linebreak elements <br> need to be inserted alongside carriage return and linefeed characters.

For some users it may be worth retroactively parsing logs in a lazy fashion (if possible, using a separate thread that operates slowly, perhaps running for 25ms, sleeping for 975, I'm unfamiliar with Pidgin's plugin API and if it will allow plugins to do this though I expect it will).

I'll be looking at fixing the issue going forward over the next week and will file a pull request if/when I do. After that's done I'll contemplate working on a log fixup function. I'll probably prefer this be something in an ini file versus a GUI function.

**Comment by warwaff** [comment]
Heya,

I'll try to have a fix ready to merge within the next week or so. I just
got a new puppy and started with a personal trainer - not much free time to
code! I've fixed outgoing messages, but have yet to get incoming messages'
linebreaks saved.

While I'm at it, there was a feature request to color chat names in emotes
based on gender. I'm still familiarizing myself with the code base, but I
may try to work on that.

Oh yeah. Derp. An external script would be a better way to do it. It'd give
me an excuse to break out Python. It shouldn't be too hard...just check for
<br> alongside LF and CRs.

**Comment by Hypertyrannical** [comment]
@vobe For this issue, I feel the best fix would be to put the text nodes into a span and use `white-space: pre;` for its CSS style, that will cause the linebreaks within it to be displayed correctly.

Please merge your fix for issue #80 when you get the chance, it's been bugging me.

**Comment by Hypertyrannical** [comment]
@vobe If you have any partially completed work on any of this, could you put up a draft PR so I could try and pick up where you left off?

**Comment by vobe** [comment]
I don't think this is an issue we can fix in the plugin, unless we split incoming and outgoing message on newlines and send then to pidgin line-by-line.  This seems to be what the IRC plugin is doing. For my take on #80, see my comments in that particular bug.
